src/java/org/apache/cassandra/schema/SchemaConstants.java

```diff
@@ -72,9 +74,31 @@ public final class SchemaConstants
 
     public static final List<String> LEGACY_AUTH_TABLES = Arrays.asList("credentials", "users", "permissions");
 
-    public static boolean isValidName(String name)
+    public static boolean isSafeLengthForFilename(String name)
+    {
+        return name.length() <= NAME_LENGTH;
+    }
+
+    /**
+     * Names such as keyspace, table, index names are used in file paths and file names,
+     * so, they need to be safe for the use there, i.e., short enough and
+     * containing only alphanumeric characters and underscores.
+     * @param name the name to check
+     * @return whether the name is safe for use in file paths and file names
+     */
+    public static boolean isNameSafeForFilename(String name)
```

Comment:
Why change the name of this public API?
I feel the new name is worse and more confusing and if I didn't know the context of this PR, I'd have trouble figuring out what the author meant by writing "Filename" here. 

The original method name was fine.

Reply:
The original name wasn't fine. As I described it led into bugs and incorrect usage of functions. The name contains the public information - it corresponds the official documentation.